### PR TITLE
Jetpack checklist: add tour

### LINKED ISSUE #33285: Jetpack checklist: ensure people know how to find back to wp-admin
Ensure folks can find their way back to wp-admin from the connect/onboarding flow.

There are currently two ways to get back:

Screenshot 2019-05-23 at 17 16 19

Could we as an MVP add some copy to some of these views and tell how to get back to wp-admin:

Screenshot 2019-05-23 at 17 14 40
Screenshot 2019-05-23 at 17 14 31
Screenshot 2019-05-23 at 17 14 23

Would it be easy to add a guided tour at the beginning of the checklist flow pointing out how to get back to wp-admin:

image

### PR DESCRIPTION
**Pull request by simison**
#### Changes proposed in this Pull Request

* Adds Jetpack checklist tour

Resolves https://github.com/Automattic/wp-calypso/issues/33285

Screenshot 2019-05-29 at 12 24 05

Screenshot 2019-05-29 at 11 59 17

#### Testing instructions

- Have a Jetpack connected site
- Open `/plans/my-plan/:site?thank-you`
- Click "continue" and you should see the checklist with two tasks:
  - One pointing to the checklist itself
  - One to the wp-admin link at the footer of the checklist

### CODE REVIEW

**Comment by simison** [comment]
@Automattic/team-calypso sorry for the ping noise — didn't mean to request a review from you all, just fat-fingered the list when creating the PR it seems. :-)

**Comment by ockham** [comment]
> @Automattic/team-calypso sorry for the ping noise — didn't mean to request a review from you all, just fat-fingered the list when creating the PR it seems. :-)

Nah they're auto-added b/c [CODEOWNERS](https://github.com/Automattic/wp-calypso/blob/ecf5f9bbc9df2eced078b6705fcfd62d2e2707e6/.github/CODEOWNERS#L35)

**Comment by simison** [comment]
>Placement of the first tour item is crappy and it's not following while scrolling.

Got this fixed, didn't depend on waiting for the element as I thought but on switching placement/target configs around. This might change if we hide the checklist at thank-you -step (https://github.com/Automattic/wp-calypso/issues/33319) but that's another worry for later.

File: client/my-sites/plans/current-plan/current-plan-thank-you-card/thank-you-card.js
Comment (ockham) [inline client/my-sites/plans/current-plan/current-plan-thank-you-card/thank-you-card.js]: Nitpick: Maybe use "mapping" fat arrow syntax:

```
	state => ( {
		siteSlug: getSelectedSiteSlug( state ),
	} ),
```

File: client/layout/guided-tours/tours/jetpack-checklist-tour/index.js
Comment (sirreal) [inline client/layout/guided-tours/tours/jetpack-checklist-tour/index.js]: Drive-by nit:

I'm rather partial to writing `Fragment`s as `<> … </>`, saves the need to add another import or reference `<React.Fragment> … </React.Fragment>`.

https://reactjs.org/docs/fragments.html#short-syntax

File: client/layout/guided-tours/tours/jetpack-checklist-tour/index.js
Comment (ockham) [inline client/layout/guided-tours/tours/jetpack-checklist-tour/index.js]:
```suggestion
							"This is your security checklist that'll help you quickly set up Jetpack. " +
```

b/c verb

File: client/layout/guided-tours/tours/jetpack-checklist-tour/index.js
Comment (ockham) [inline client/layout/guided-tours/tours/jetpack-checklist-tour/index.js]: Might add to the WP vs WP.com confusion? :sob:

File: client/layout/guided-tours/tours/jetpack-checklist-tour/index.js
Comment (simison) [reply client/layout/guided-tours/tours/jetpack-checklist-tour/index.js]: I prefer `<>` myself as well but I also prefer consistency over anything else, so I've stuck to Fragment in Calypso until now. Looks like  `<>` has crept up in numbers tho so ok to switch. :-)

File: client/layout/guided-tours/tours/jetpack-checklist-tour/index.js
Comment (simison) [reply client/layout/guided-tours/tours/jetpack-checklist-tour/index.js]: Haha, yeah. I don't really think "WP Admin" is clear for the average user either but maybe something like  "self-hosted WordPress dashboard" as we have in the checklist footer?

image

Comment (ockham) [reply client/layout/guided-tours/tours/jetpack-checklist-tour/index.js]: This step's bubble is off-screen for me (13"). It'd be nice to scroll down to it. There might be a Guided Tour prop to implement that: `scrollContainer` (see https://github.com/Automattic/wp-calypso/blob/8d77bf347ec1065c9a1833dac94e549a430190ff/client/layout/guided-tours/docs/API.md#props-1)

**Review by ockham — APPROVED** [review]
Looking good overall, and testing well. Left a few nits. UX-wise, it'd be nice to scroll to the wp-admin button step: https://github.com/Automattic/wp-calypso/pull/33373/files#r288481080

Feel free to :ship: after addressing (or ignoring :stuck_out_tongue: )

File: client/my-sites/plans/current-plan/current-plan-thank-you-card/thank-you-card.js
Comment (simison) [reply client/my-sites/plans/current-plan/current-plan-thank-you-card/thank-you-card.js]: Sure!

File: client/layout/guided-tours/tours/jetpack-checklist-tour/index.js
Comment (ockham) [reply client/layout/guided-tours/tours/jetpack-checklist-tour/index.js]: > Haha, yeah. I don't really think "WP Admin" is clear for the average user either but maybe something like "self-hosted WordPress dashboard" as we have in the checklist footer?
> 
> image

While `self-hosted WordPress dashboard` is more explicit, I'm still worried about mentioning the `WordPress.com dashboard` in the same sentence, this early during the setup process.

I mean it's conceivable that people making it to this step are all power users, and that they've gone through wp-admin and WP.com (for JP connect) anyway, but maybe we can still avoid mentioning `WordPress` and `WordPress.com` in the same sentence?

Maybe just refer to the `WordPress.com dashboard` as `this dashboard`? I.e. `return to your WordPress [admin panel? dashboard? installation?] or continue using this dashboard`?

File: client/layout/guided-tours/tours/jetpack-checklist-tour/index.js
Comment (ockham) [reply client/layout/guided-tours/tours/jetpack-checklist-tour/index.js]: I think `<Fragment />` has been lingering from times when our toolchain (babel upgrade blocked by eslint upgrade or whatever) wasn't able to handle `</>` just yet.

Thus +1 on using `</>` in newly written code (and replacing `<Fragment />` in legacy code that we touch.)

File: client/layout/guided-tours/tours/jetpack-checklist-tour/index.js
Comment (simison) [reply client/layout/guided-tours/tours/jetpack-checklist-tour/index.js]: I'll switch to "this dashboard" for now, good call. I'll make a point to follow up with @jeffgolenski next week on this. I also want to revisit the "thank-you" -modal copy and the footer text all at once to make sure it's all clear and easy to understand.

**Comment by simison** [comment]
>This step's bubble is off-screen for me (13"). It'd be nice to scroll down to it. There might be a Guided Tour prop to implement that: scrollContainer (see https://github.com/Automattic/wp-calypso/blob/8d77bf347ec1065c9a1833dac94e549a430190ff/client/layout/guided-tours/docs/API.md#props-1)

I've tested `scrollContainer`  with `.jetpack-checklist__footer`, `body` and `html`. With all those, the step is very buggy. It can hide out somewhere until the window is resized, doesn't get connected to the right element and thus follow it when scrolling etc..

Let's revisit that later on after merging — seems like it's a bug in the framework, I'll open an issue to track.

Feels weird that the framework just doesn't take care of it on its own instead of us manually defining this behavior...

File: client/layout/guided-tours/tours/jetpack-checklist-tour/index.js
Comment (simison) [reply client/layout/guided-tours/tours/jetpack-checklist-tour/index.js]: Done!

File: client/my-sites/plans/current-plan/current-plan-thank-you/thank-you.js
Comment (simison) [inline client/my-sites/plans/current-plan/current-plan-thank-you/thank-you.js]: @sirreal I switched this back to class Component in order to get `requestGuidedTour` functioning. Lemme know if there's another way. :-)

**Comment by simison** [comment]
Rebased! I'll merge little later today.

Feel free to have a final look meanwhile but not much changed since the last ✅.

File: client/my-sites/plans/current-plan/current-plan-thank-you/thank-you.js
Comment (sirreal) [reply client/my-sites/plans/current-plan/current-plan-thank-you/thank-you.js]: There are other ways, each with their own tradeoffs. This approach is perfectly fine, I'm not too concerned with whether we have function or class components generally.

**Comment by sirreal** [comment]
> > This step's bubble is off-screen for me (13"). It'd be nice to scroll down to it. There might be a Guided Tour prop to implement that: scrollContainer (see https://github.com/Automattic/wp-calypso/blob/8d77bf347ec1065c9a1833dac94e549a430190ff/client/layout/guided-tours/docs/API.md#props-1)
> 
> I've tested `scrollContainer` with `.jetpack-checklist__footer`, `body` and `html`. With all those, the step is very buggy. It can hide out somewhere until the window is resized, doesn't get connected to the right element and thus follow it when scrolling etc..
> 
> Let's revisit that later on after merging — seems like it's a bug in the framework, I'll open an issue to track.
> 
> Feels weird that the framework just doesn't take care of it on its own instead of us manually defining this behavior...

Try setting `shouldScrollTo={ true }` on the Step? Seems to be an undocumented prop…

**Comment by simison** [comment]
>Try setting shouldScrollTo={ true } on the Step? Seems to be an undocumented prop…

No dice. :-( A bit better tho! It shows up in the wrong place but attaches correctly after `onScroll` event.

image

**Comment by simison** [comment]
Actually combination of `scrollContainer="html" shouldScrollTo` gets us _almost_ there:
- it scrolls to right place
- the arrow is pointing to right element
- ...but when I scroll, the step doesn't stick to the element :-(
    Screenshot 2019-05-31 at 15 34 24

I'll just ship this without those and we can loop back to fixing this stuff later.

**Comment by sirreal** [comment]
> * ...but when I scroll, the step doesn't stick to the element :-(

There's also prop: `keepRepositioning`, sounds worth a try

**Comment by simison** [comment]
>There's also prop: keepRepositioning, sounds worth a try

Gets us even closer to success but still nada. :-( Follow-up: https://github.com/Automattic/wp-calypso/pull/33506 — I'll elaborate there more next week.

**Pull request merged**
